### PR TITLE
Add gift cards page header and table

### PR DESCRIPTION
**Pull request by mmarkusik**
I want to merge this change because it adds gift cards list page header and table. Since api is not available yet, this is very much a WIP, hence some stuff is commented out, not typed properly etc. 

https://app.clickup.com/t/2549495/SALEOR-3989

**PR intended to be tested with API branch:** 

### Screenshots

Screenshot 2021-07-20 at 12 46 39

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://qa.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3942 Add Gift Cards list page header & table](https://app.clickup.com/t/2549495/SALEOR-3942)

**Comment by patrys** [comment]
Task linked: [SALEOR-3989 Create components for gift cards list table](https://app.clickup.com/t/2549495/SALEOR-3989)

File: src/hooks/useBulkActions.ts [hunk 1]
```diff
@@ -31,3 +31,14 @@ function useBulkActions(initial: string[] = []) {
...
+export interface BulkActions {
+  add: (data: string) => void;
+  isSelected: (data: string) => boolean;
+  listElements: string[];
+  remove: (data: string) => void;
+  reset: () => void;
+  set: (data: string[]) => void;
+  toggle: (data: string) => void;
+  toggleAll: (itmes: Node[], selected: number) => void;
+}
```
Comment (orzechdev) [inline src/hooks/useBulkActions.ts:44]: I don't see this interface is used at all.

File: src/giftCards/GiftCardsList/GiftCardsList.tsx [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+import GiftCardsListTable from "./GiftCardsListTable";
+import { GiftCardListUrlQueryParams } from "./types";
+
+interface GiftCardsListProps {
+  params: GiftCardListUrlQueryParams;
+}
+
+const GiftCardsList: React.FC<GiftCardsListProps> = ({ params }) => {
+  return (
+    <GiftCardsListProvider params={params}>
```
Comment (orzechdev) [inline src/giftCards/GiftCardsList/GiftCardsList.tsx:15]: 👍

**Review by orzechdev — CHANGES_REQUESTED** [review]

File: src/hooks/useBulkActions.ts (hunk 1 above)
Comment (mmarkusik) [reply src/hooks/useBulkActions.ts]: wip leftover, removing

**Review by orzechdev — APPROVED** [review]

**Pull request merged**